### PR TITLE
Apply ruff/pyupgrade rules (UP)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
```
UP030 Use implicit references for positional format fields
```
https://docs.astral.sh/ruff/rules/format-literals/

```
UP031 Use format specifiers instead of percent format
```
https://docs.astral.sh/ruff/rules/printf-string-formatting/
```
UP032 Use f-string instead of `format` call
```
https://docs.astral.sh/ruff/rules/f-string/

### CODE REVIEW

**Review by pradyunsg — COMMENTED** [review]
I don't think UP032 is a good rule to be enforcing as-is, at least not without changing the surrounding logic to keep it valid.

File: tests/test_requirements.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def test_basic_valid_requirement_parsing(
     parts = [name]
     if extras:
         parts.append("[")
-        parts.append("{ws},{ws}".format(ws=whitespace).join(sorted(extras)))
+        parts.append(f"{whitespace},{whitespace}".join(sorted(extras)))
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply tests/test_requirements.py:121]: I see no easy alternative other than disabling `UP032`? Locally with a `noqa:`? Globally?

File: src/packaging/tags.py [hunk 2]
```diff
@@ -467,20 +462,15 @@ def mac_platforms(
...
         else:
             for minor_version in range(16, 3, -1):
                 compat_version = 10, minor_version
                 binary_format = "universal2"
-                yield "macosx_{major}_{minor}_{binary_format}".format(
-                    major=compat_version[0],
-                    minor=compat_version[1],
-                    binary_format=binary_format,
-                )
+                yield f"macosx_{compat_version[0]}_{compat_version[1]}_{binary_format}"
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply src/packaging/tags.py:473]: Would you like to disable `UP032`? Or perhaps change to the following?
```python
                major=compat_version[0]
                minor=compat_version[1]
                yield f"macosx_{major}_{minor}_{binary_format}"
```

File: tests/test_requirements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply tests/test_requirements.py:121]: But then the `extras` case **is** different from the rest.

File: src/packaging/tags.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply src/packaging/tags.py]: I tried a variant of the second suggestion, which results in more consistent code — and perhaps more readable.

File: tasks/check.py [hunk 3]
```diff
@@ -107,35 +107,22 @@ def pep440(cached=False):
...
+        f"{len(data)} ({len(only_invalid_versions) / len(data):.2%})"
     )
     print(
-        "Projects with Differing Latest Version:   {}/{} ({:.2%})".format(
-            len(differing_latest_versions),
-            len(data),
-            len(differing_latest_versions) / len(data),
-        )
+        f"Projects with Differing Latest Version:   {len(differing_latest_versions)}/"
+        f"{len(data)} ({len(differing_latest_versions) / len(data):.2%})"
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply tasks/check.py:127]: I can understand why you prefer the first form. Add a `noqa:` or disable `UP032` globally?

**Review by brettcannon — APPROVED** [review]
I'm fine w/ the changes if @pradyunsg is.

File: tests/test_requirements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply tests/test_requirements.py]: I disabled `UP032` globally.

File: tasks/check.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply tasks/check.py]: I eventually disabled `UP032` globally.

**Review by pradyunsg — APPROVED** [review]

**Comment by pradyunsg** [comment]
> Fatal Python error: Segmentation fault

Wow.

**Comment by brettcannon** [comment]
Thanks!

**Pull request merged**